controller/src/api/mod.rs

```diff
@@ -20,45 +23,25 @@ impl From<i32> for Crud {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum RikError {
+    #[error("IO error: {0}")]
     IoError(std::io::Error),
-    HttpRequestError(serde_json::Error),
+
+    #[error("HTTP request error: {0}")]
+    ParsingError(serde_json::Error),
+
+    #[error("Internal communication error: {0}")]
     InternalCommunicationError(String),
```

Comment:
I don't think we should be using `thiserror` for public facing APIs, we expect APIs to be able to return JSON and a proper HTTP status, not only a string. Also, ownership of HTTP status should stay at API methods level and not be handle through a `From<>` trait implementation, as we were doing previously. 

`thiserror` should only used for communication between components (controller, scheduler, riklet) and internal business components (`core-:>instance_service`)

We can discuss about it if you want.

Reply:
Nothing to discuss I also think the same, I do not know why I did that